File: 06_SpringBoot/rh/src/main/java/com/avanade/decolatech/rh/entities/Candidato.java
package com.avanade.decolatech.rh.entities;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;

@Entity
@Table(name = "TB_CANDIDATOS")
public class Candidato {
	
	@Id
	@Column(name = "CPF")
	private String cpf;
	
	@Column(name = "NOME")
	private String nome;
	
	@Column(name = "TELEFONE")
	private String telefone;
	
	@Column(name = "EMAIL")
	private String email;
	
	@JsonIgnore
	@OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, mappedBy = "candidato")
	private List<Inscricao> inscricoes;

	public String getCpf() {
		return cpf;
	}

	public void setCpf(String cpf) {
		this.cpf = cpf;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getTelefone() {
		return telefone;
	}

	public void setTelefone(String telefone) {
		this.telefone = telefone;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public List<Inscricao> getInscricoes() {
		return inscricoes;
	}

	public void setInscricoes(List<Inscricao> inscricoes) {
		this.inscricoes = inscricoes;
	}
	
}

File: 07_Angular/app-angular-rh/src/app/classes/candidato.ts
export class Candidato {

    constructor(
        public cpf: string = '', 
        public nome: string = '', 
        public telefone: string = '', 
        public email: string = '') {}

    // constructor() {
    //     this._cpf='';
    //     this._nome='';
    //     this._telefone='';
    //     this._email='';
    // }
    
    // _cpf : string;
    // public get cpf() : string {
    //     return this._cpf;
    // }
    // public set cpf(v : string) {
    //     this._cpf = v;
    // }
    
    
    // _nome : string;
    // public get nome() : string {
    //     return this._nome;
    // }
    // public set nome(v : string) {
    //     this._nome = v;
    // }
    
    
    // _telefone : string;
    // public get telefone() : string {
    //     return this._telefone;
    // }
    // public set telefone(v : string) {
    //     this._telefone = v;
    // }
    
    
    // _email : string;
    // public get email() : string {
    //     return this._email;
    // }
    // public set email(v : string) {
    //     this._email = v;
    // }
    
    public mostrar() : string {
        return `CPF: ${this.cpf}\nNome: ${this.nome}\nFone: ${this.telefone}\nEmail: ${this.email}`;
    }
}
